samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -0,0 +1,244 @@
+from .pipeline_templates_manifest import PipelineTemplateManifest, PipelineTemplatesManifest
+
+LOG = logging.getLogger(__name__)
+shared_path: Path = global_cfg.config_dir
+APP_PIPELINE_TEMPLATES_REPO_URL = "https://github.com/aws/aws-sam-cli-pipeline-init-templates.git"
+APP_PIPELINE_TEMPLATES_REPO_LOCAL_NAME = "aws-sam-cli-app-pipeline-templates"
+CUSTOM_PIPELINE_TEMPLATE_REPO_LOCAL_NAME = "custom-pipeline-template"
+
+
+def do_interactive() -> None:
+    """
+    An interactive flow that prompts the user for pipeline template (cookiecutter template) location, downloads it,
+    runs its specific questionnaire then generates the pipeline config file based on the template and user's responses
+    """
+    click.echo("Which pipeline template source would you like to use?")
+    click.echo("\t1 - AWS Quick Start Pipeline Templates\n\t2 - Custom Pipeline Template Location")
+    location_choice = click.prompt("Choice", type=click.Choice(["1", "2"]), show_choices=False)
+    if location_choice == "2":
+        _generate_from_custom_location()
+    else:
+        _generate_from_app_pipeline_templates()
+
+
+def _generate_from_app_pipeline_templates() -> None:
+    """
+    Prompts the user to choose a pipeline template from SAM predefined set of pipeline templates hosted in the git
+    repository: aws/aws-sam-cli-pipeline-init-templates.git
+    downloads locally, then generates the pipeline config file from the selected pipeline template.
+    """
+    pipeline_templates_local_dir: Path = _clone_app_pipeline_templates()
+
+    pipeline_templates_manifest: PipelineTemplatesManifest = _read_app_pipeline_templates_manifest(
+        pipeline_templates_dir=pipeline_templates_local_dir
+    )
+    # The manifest contains multiple pipeline-templates so select one
+    selected_pipeline_template_manifest: PipelineTemplateManifest = _select_pipeline_template(
+        pipeline_templates_manifest=pipeline_templates_manifest
+    )
+    selected_pipeline_template_dir: Path = pipeline_templates_local_dir.joinpath(
+        selected_pipeline_template_manifest.location
+    )
+    _generate_from_pipeline_template(pipeline_template_dir=selected_pipeline_template_dir)
+
+
+def _generate_from_custom_location() -> None:
+    """
+    Prompts the user for a custom pipeline template location, downloads locally, then generates the pipeline config file
+    """
+    pipeline_template_repo_url: str = click.prompt(text="template Git location")
+    pipeline_template_local_dir: Path = _clone_custom_pipeline_template(repo_url=pipeline_template_repo_url)
+    _generate_from_pipeline_template(pipeline_template_dir=pipeline_template_local_dir)
+    # Unlike app pipeline templates, custom pipeline templates are not shared between different SAM applications
+    # and should be cleaned up from users' machines after generating the pipeline config files.
+    shutil.rmtree(pipeline_template_local_dir, onerror=rmtree_callback)
```

Comment:
Is `pipeline_template_local_dir` outside of the SAM app root directory? What's different about custom templates?

Reply:
it is outside the SAM app; in a shared location (default ~/.aws-sam/). 
[App pipeline templates](https://github.com/aws/aws-sam-cli-pipeline-init-templates), are the ones owned by SAM and can be shared between different SAM projects.

custom ones are provided by the user and the assumption is that it is not shared between different SAM projects.

Reply:
I see. Why do we store them in a shared directory?

Reply:
We are sure the user has read/write permissions to this shared directory (otherwise we wouldn't be able to create the SAM project). But actually it make more sense to store in `/tmp`

Reply:
Why not store in `.aws-sam` in the SAM project? All projects should be well isolated.

Reply:
storing in /tmp will automatically clean it up rather thank manually remove it.

Reply:
Isn't the `.aws-sam` directory already to store all kinds of project-related semi-temporary files? What value does another directory add?

Reply:
I don't have a strong opinion here. The only difference between this case and others is that we cleanup the pipeline template after usage. So, I think it is fine to clone to the project's `.aws-sam` folder and keep it without cleanup.